### DISCUSSION TITLE
How to insert f64 type?

### DISCUSSION THREAD

**Discussion by jjcfrancisco** [rusqlite/rusqlite#1439] [Questions]
Hi, it might be me, but I cannot find how to insert values of `f64`? `item.length` is a `f64` but I get a compiler error:

```bash
mismatch types expected &i64 found &f64
```

Function:

```rust
fn insert_to_db(data: HashMap<i64, Prop>) {
    let conn = Connection::open("./my_data.db").expect("Error opening db");
    conn.execute(
        "CREATE TABLE my_table (
            id INTEGER AUTO_INCREMENT PRIMARY KEY,
            property_id INTEGER,
            length REAL,
        )",
        (),
    )
    .expect("Error creating table");

    for (_, item) in data.iter() {
        conn.execute(
            "INSERT INTO my_table (property_id, length) VALUES (?1, ?2)",
            [&item.prop_id, &item.length],
        ).expect("Error inserting data");
    }
}
```

**Comment by gwenn** [comment]
When you want to insert values which types are heterogeneous, you must use tuples (and not arrays which need homogeneous type).
So I guess you just need to replace `[&item.prop_id, &item.length]` by `(&item.prop_id, &item.length)`.
See https://docs.rs/rusqlite/latest/rusqlite/trait.Params.html#passing-parameters-to-sqlite for more details.
